grpcutil/dns_resolver.go

```diff
@@ -101,8 +101,11 @@ func parseTarget(target string) (host, port string, err error) {
 	return "", "", fmt.Errorf("invalid target address %v", target)
 }
 
-// Resolve creates a watcher that watches the name resolution of the target.
-func (r *dnsResolver) Resolve(target string) (Watcher, error) {
+// Resolve creates a watcher that watches the SRV/hostname record resolution of the target.
+//
+// If service is not empty, the watcher will first attempt to resolve an SRV record.
```

Comment:
This description is missing one more case:
```
// If target can be parsed as IP address, watcher will return it, and will not send any more updates afterwards.
```

Reply:
Done.